MLDSAI/OpenAdapt
# update dependencies with SAM and gitignore

**What kind of change does this PR introduce?**

This PR introduces two changes: adding *pth gitignore and adding "sam" to requirements.txt.

**Summary**

The motivation behind this change is to improve the project's development environment. By adding the "*.pth" file to the gitignore, we ensure that these files, which can be generated during the development process, are not accidentally committed to the repository. Additionally, by adding "sam" to the requirements.txt file, we include the "sam" package as a project dependency, which is required for specific functionality.

**Checklist**
* [x] I have perfomed a self-review of my code

**How can your code be run and tested?**

To test the changes, follow these steps:

Update your local copy of the repository.
python -m openadapt.record "test"
python -m openadapt.replay SamReplayStrategy
Ensure that the "*.pth" files are ignored and not included in any commits.
Verify that "sam" is added as a dependency in the requirements.txt file.
